src/engine/controls/cuecontrol.cpp

```diff
@@ -366,42 +363,63 @@ void CueControl::trackLoaded(TrackPointer pNewTrack) {
         CuePointer pCue(pNewTrack->createAndAddCue());
         pCue->setPosition(cuePoint.getPosition());
         pCue->setHotCue(-1);
-        pCue->setType(Cue::LOAD);
+        pCue->setType(Cue::Type::MainCue);
     }
     m_pCuePoint->set(cuePoint.getPosition());
 
     // Update COs with cues from track.
     loadCuesFromTrack();
 
     // Seek track according to SeekOnLoadMode.
-    SeekOnLoadMode seekOnLoadMode = getSeekOnLoadMode();
+    SeekOnLoadMode seekOnLoadMode = getSeekOnLoadPreference();
+
+    CuePointer pAudibleSound = pNewTrack->findCueByType(Cue::Type::AudibleSound);
+    double firstSound = kNoTrigger;
+    if (pAudibleSound) {
+        firstSound = pAudibleSound->getPosition();
+    }
+
+    double mainCue = m_pCuePoint->get();
+    if (m_pConfig->getValue(ConfigKey("[Controls]", "MoveIntroStartWithMainCue"), false)) {
+        if (mainCue != kNoTrigger && mainCue != 0) {
+            m_pIntroStartPosition->set(mainCue);
```

Comment:
I originally proposed this but you suggested to combine the options for simplicity of the UI. I am okay with separating them.

Reply:
Due to the different defaults demand I now think separate flags are better.
We even considder to hide the flag for migration. The important part here is that the migration becomes a one time thing for each track. This can be controlled by the presence of IntroStart and main cue not at 0:00

Reply:
I am now wondering, do we really need the "move intro start with main cue" feature at all? Previously I thought it could be a convenience for users who want to set the intro start but don't have the intro & outro cues mapped on their controller. On further thought though, I don't think there would be much of a purpose to this if the user is not also setting the intro end and outro start points manually, in which case their controller mapping would likely include all the intro & outro cues. I think we only need the option for the analyzer to set the intro start to the main cue point once for migration from earlier Mixxx versions.

Reply:
I went ahead and removed the "move intro start with main cue" functionality while moving the initial placement of the intro start point to the old main cue point to AnalyzerSilence: 
18b13f9